flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/AggregatedStatistics.java

```diff
@@ -35,28 +35,28 @@ class AggregatedStatistics implements Serializable {
   private final long checkpointId;
   private final StatisticsType type;
   private final Map<SortKey, Long> keyFrequency;
-  private final SortKey[] rangeBounds;
+  private final SortKey[] keySamples;
```

Comment:
this rename is needed so that `AggregatedStatistics` can be used to store both the complete samples and calculated range bounds.

Reply:
Would it be better to have a separate class for global stat, and aggregate stat? It's a bit late here, and it's a bit hard to follow when the keys keySamples are actually range bounds, and when they contain the full data...
Maybe tomorrow it will be easier to follow 😀

Reply:
that is definitely a fair question. I also thought about it. I can make the change if you think it is better.

Reply:
Let's talk about this offline. I don't see all the cons and pros ATM.

Reply:
Reading through the code again, I'm more-and-more convinced, that we have 2 different objects here:
- RangeBounds (former global statistics) - key-values of the weights used by the partitioner with hash
- Statistics (former completed statistics) - Sketch or Map without hash, but full of data

I think we are just confusing them because of historical reasons.

Reply:
Yes, conceptually we have two types of objects here. for Map statistics, there is no difference btw global statistics and completed statistics, as there is no further reduction in stats size. For sketch statistics, global statistics is a lot smaller with range bounds.

We can introduce two types `CompleteStatistics` and `GlobalStatistics`. We can also introduce a base type `AggregatedStatistics`. I am trying to avoid duplicate the `AggregatedStatisticsSerializer` as it can work for both types. Maybe generics can enable code reused and solve the most duplications.

Reply:
last commit separated out the two types

Reply:
I have removed the `AggregatedStatistics` base class. With the change to the `GlobalStatistics` on Map key assignment, the base class doesn't make much sense anymore. Now `GlobalStatistics` is used by partitioner and `CompletedStatistics` are raw aggregated stats.